src/pattern_matching/pssm/dnamotif.rs

```diff
@@ -0,0 +1,269 @@
+                }
+            }
+        }
+        let mut m = DNAMotif {
+            seq_ct: seqs.len(),
+            scores: counts,
+            min_score: 0.0,
+            max_score: 0.0,
+        };
+        m.normalize();
+        m.calc_minmax();
+        Ok(m)
+    }
+
+    // helper function -- normalize self.scores
+    fn normalize(&mut self) {
+        for i in 0..self.len() {
+            let mut tot: f32 = 0.0;
+            // FIXME: slices would be cleaner
+            for base_i in 0..4 {
+                tot += self.scores[[i, base_i]];
+            }
+            for base_i in 0..4 {
+                self.scores[[i, base_i]] /= tot;
+            }
+        }
+    }
+
+    // helper function
+    pub fn calc_minmax(&mut self) {
+        let pssm_len = self.len();
+
+        // score corresponding to sum of "worst" bases at each position
+        // FIXME: iter ...
+        self.min_score = 0.0;
+        for i in 0..pssm_len {
+            // can't use the regular min/max on f32, so we use f32::min
+            let min_sc = (0..4).map(|b| self.scores[[i, b]]).fold(INFINITY, f32::min);
+            self.min_score += min_sc;
+        }
+
+        // score corresponding to "best" base at each position
+        self.max_score = 0.0;
+        for i in 0..pssm_len {
+            let max_sc = (0..4)
+                .map(|b| self.scores[[i, b]])
+                .fold(NEG_INFINITY, f32::max);
+            self.max_score += max_sc;
+        }
+    }
+}
+
+impl Motif for DNAMotif {
+    const LK: [u8; 127] = [
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 0, 255, 3, 255, 255, 255, 2, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 1, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 0, 255, 3, 255, 255, 255, 2, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 1, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    ];
+    const MONOS: &'static [u8] = b"ATGC";
+
+    fn rev_lk(idx: usize) -> u8 {
+        match idx {
+            0 => b'A',
+            1 => b'T',
+            2 => b'G',
+            3 => b'C',
+            _ => INVALID_MONO,
+        }
+    }
+
+    fn len(&self) -> usize {
+        self.scores.dim().0
+    }
+
+    fn get_scores(&self) -> &Array2<f32> {
+        &self.scores
+    }
+    fn get_min_score(&self) -> f32 {
+        self.min_score
+    }
+    fn get_max_score(&self) -> f32 {
+        self.max_score
+    }
+    fn get_bits() -> f32 {
+        2.0
+    }
+
+    /// derived from
+    /// https://github.com/biopython/biopython/blob/master/Bio/motifs/matrix.py#L205
+    fn degenerate_consensus(&self) -> Result<Vec<u8>, PSSMError> {
```

Comment:
Change this to a normal comment inside the function. Otherwise, it would overwrite the doc comment of the trait.